AdventOfCode2016/Day1.fs

```diff
@@ -0,0 +1,84 @@
+﻿module Day1
+
+open System
+open System.IO
+
+type Turn = Left | Right
+
+type Direction = North | East | West | South
+
+type Instruction = {
+    Direction: Turn;
+    Distance: int;
+}
+
+type Coordinate = {
+    X : int;
+    Y : int;
+}
+
+type State = {
+    Position: Coordinate;
+    Facing: Direction;
+}
+
+let parseDirection (c:char) =
+    match c with
+    | 'L' -> Left
+    | 'R' -> Right
+    | _ -> raise <| new System.FormatException(sprintf "Could not parse %c" c)
+
+let parseInstruction (s:string) =
+    {
+        Direction = parseDirection s.[0]
+        Distance = Int32.Parse(s.Substring(1))
+    }
+
+let left facing =

```

Comment:
As a note, you always want the 'state' parameters to be after 'argument' parameters in functional languages. This allows for the clean syntax / piping you expect. Since `facing` is a state variable, it belongs after `direction`, which is an argument to permute the state.